dot/parachain/backing/validated_candidate_command.go

```diff
@@ -40,5 +99,252 @@ const (
+		CandidateHash:             candidateHash,
+		CommittedCandidateReceipt: commitedCandidate,
+		PersistedValidationData:   pvd,
+	}
+
+	// sanity check that we're allowed to second the candidate.
+	fragmentTreeMembership, err := cb.secondingSanityCheck(hypotheticalCandidate, false)
+	if err != nil {
+		return fmt.Errorf("not allowed to second: %w", err)
+	}
+
+	statement := parachaintypes.NewStatementVDT()
+	err = statement.Set(parachaintypes.Seconded(commitedCandidate))
+	if err != nil {
+		return fmt.Errorf("setting statement: %w", err)
+	}
+
+	// If we get an errRejectedByProspectiveParachains,
+	// then the statement has not been distributed or imported into the table
+	signedFullStatementWithPVD, err := signImportAndDistributeStatement(
+		cb.SubSystemToOverseer, rpState, cb.perCandidate, statement, &pvd, cb.keystore)
+
+	if err != nil {
+		if errors.Is(err, errRejectedByProspectiveParachains) {
+			cb.SubSystemToOverseer <- collatorprotocolmessages.Invalid{
+				Parent:           candidate.Descriptor.RelayParent,
+				CandidateReceipt: candidate,
+			}
+		}
+		return err
+	}
+
+	perCandidate, ok := cb.perCandidate[candidateHash]
+	if !ok {
+		logger.Warnf("missing `per candidate` for seconded candidate: %s", candidateHash.Value)
+	} else {
+		perCandidate.secondedLocally = true
+	}
+
+	// update seconded depths in active leaves.
+	for leaf, depths := range fragmentTreeMembership {
+		leafState, ok := cb.perLeaf[leaf]
+		if !ok {
+			logger.Warnf("missing `per leaf` for known active leaf: %s", leaf)
+			continue
+		}
+
+		secondedAtDepth, ok := leafState.secondedAtDepth[parachaintypes.ParaID(candidate.Descriptor.ParaID)]
+		if !ok {
+			var btreeMap btree.Map[uint, parachaintypes.CandidateHash]
+			leafState.secondedAtDepth[parachaintypes.ParaID(candidate.Descriptor.ParaID)] = &btreeMap
+			secondedAtDepth = &btreeMap
+		}
+
+		for _, depth := range depths {
+			secondedAtDepth.Set(depth, candidateHash)
+		}
+	}
+
+	rpState.issuedStatements[candidateHash] = true
+	cb.SubSystemToOverseer <- collatorprotocolmessages.Seconded{
+		Parent: rpState.relayParent,
+		Stmt:   signedFullStatementWithPVD.SignedFullStatement,
+	}
+
+	return nil
+}
+
+func (cb *CandidateBacking) handleCommandAttest(
+	bgValidationResult backgroundValidationResult,
+	candidateHash parachaintypes.CandidateHash,
+	rpState *perRelayParentState,
+) error {
+	// We are done - no need to validate this candidate again.
+	delete(rpState.fallbacks, candidateHash)
+
+	// sanity check.
+	if rpState.issuedStatements[candidateHash] {
+		// already issued a statement for this candidate
+		return nil
+	}
+
+	statement := parachaintypes.NewStatementVDT()
+	if err := statement.Set(parachaintypes.Valid(candidateHash)); err != nil {
+		return fmt.Errorf("setting statement: %w", err)
+	}
+
+	if bgValidationResult.err == nil {
+		if _, err := signImportAndDistributeStatement(
+			cb.SubSystemToOverseer, rpState, cb.perCandidate, statement, nil, cb.keystore,
+		); err != nil {
+			return err
```

Comment:
wrap this error?

Reply:
`signImportAndDistributeStatement`will return a wrapped error.
- [error while signing ](https://github.com/ChainSafe/gossamer/blob/cb9c5735d385da3d605c50bc9220902098302dfb/dot/parachain/backing/validated_candidate_command.go#L329)
- [error while importing statements](https://github.com/ChainSafe/gossamer/blob/cb9c5735d385da3d605c50bc9220902098302dfb/dot/parachain/backing/validated_candidate_command.go#L339)
- for statement distribution, we send a message to the overseer. Also, this message returns nothing. So, there is no error in this action